src/test/java/org/beanpod/switchboard/fixture/EncoderFixture.java

```diff
@@ -1,33 +1,46 @@
 package org.beanpod.switchboard.fixture;
 
-import org.beanpod.switchboard.entity.ChannelEntity;
-import org.beanpod.switchboard.entity.DeviceEntity;
-import org.beanpod.switchboard.entity.EncoderEntity;
+import net.bytebuddy.asm.Advice;
+import org.beanpod.switchboard.dto.EncoderDTO;
+import org.beanpod.switchboard.entity.*;
+import org.beanpod.switchboard.dto.EncoderDTO;
+import org.openapitools.model.EncoderModel;
 
+import java.beans.Encoder;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
 public class EncoderFixture {
-
+    public static final String  SERIAL_NUMBER = "1";
     private static String pattern = "yyyy-MM-dd HH:mm:ss";
     private static SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern);
-    static private Set<ChannelEntity> setOfChannels = ChannelFixture.getSetOfChannels();
+    static private Set<OutputChannelEntity> setOfChannels = ChannelFixture.getOutputChannelEntities();
 
-    public static EncoderEntity getEncoder1(DeviceEntity device) throws ParseException {
-        return new EncoderEntity("1",
-                simpleDateFormat.parse("2020-10-31 05:05:05"),
-                device,
-                setOfChannels);
+    public static EncoderEntity getEncoder1(DeviceEntity device){
+        try {
+            return new EncoderEntity(SERIAL_NUMBER,
+                    simpleDateFormat.parse("2020-10-31 05:05:05"),
+                    device,
+                    setOfChannels);
+        } catch (ParseException e) {
+            e.printStackTrace();
+            throw new RuntimeException("Unable to parse the date in Encoder 1");
+        }
```

Comment:
I guess this is because ParseException is a checked exception? Cause it's kind of weird to have a try catch in a fixture since it should always work

Reply:
Yes, the `simpleDateFormat.parse` throws that checked exception. I decided to catch it here instead of propagating it all de way to tests. Anyways it boils down to pretty much the same result.